pkg/webhooks/server.go

```diff
@@ -372,6 +377,11 @@ func (ws *WebhookServer) ResourceMutation(request *v1beta1.AdmissionRequest) *v1
 
 func (ws *WebhookServer) resourceValidation(request *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse {
 	logger := ws.log.WithName("Validate").WithValues("uid", request.UID, "kind", request.Kind.Kind, "namespace", request.Namespace, "name", request.Name, "operation", request.Operation)
+	if request.Operation == v1beta1.Delete {
+		ws.handleDelete(request)
+	} else if request.Operation == v1beta1.Update {
+		ws.handleUpdate(request)
+	}
```

Comment:
Can we move this logic to `ResourceMutation`? Ideally we want all generate logic in `HandleGenerate`. https://github.com/kyverno/kyverno/blob/58feb4f0aeeb27dd4a400ddaa815ba97b00247eb/pkg/webhooks/server.go#L355-L360

Reply:
I have added enqueuing of GR on updating of clone source in `ResourceMutation`. 

But when the source is deleted, resource mutation is not called. So, the delete logic is handled by `resourceValidation`.